lib/response.js

```diff
@@ -81,11 +81,11 @@ module.exports = {
 
   set status(code) {
     assert('number' == typeof code, 'status code must be a number');
-    assert(statuses[code], `invalid status code: ${code}`);
+    assert(statuses[code] || this.customStatuses[code], `invalid status code: ${code}`);
     assert(!this.res.headersSent, 'headers have already been sent');
     this._explicitStatus = true;
     this.res.statusCode = code;
-    if (this.req.httpVersionMajor < 2) this.res.statusMessage = statuses[code];
+    if (this.req.httpVersionMajor < 2) this.res.statusMessage = statuses[code] || this.customStatuses[code];
```

Comment:
If I get you correctly, you meant to use the custom status code instead of the default status code. Right? So, actually the logic would be: 
```
this.customStatuses[code] || statuses[code];
```
Then, if you have no custom status codes the default status will be used instead.